MapleServer2/Managers/UGCMarketManager.cs

```diff
@@ -0,0 +1,149 @@
+﻿using Maple2Storage.Enums;
+using Maple2Storage.Tools;
+using MapleServer2.Data.Static;
+using MapleServer2.Database;
+using MapleServer2.Enums;
+using MapleServer2.PacketHandlers.Game.Helpers;
+using MapleServer2.Types;
+
+namespace MapleServer2.Managers;
+
+public class UGCMarketManager
+{
+    private readonly Dictionary<long, UGCMarketItem> Items;
+    private readonly Dictionary<long, UGCMarketSale> Sales;
+
+    public UGCMarketManager()
+    {
+        Items = new();
+        List<UGCMarketItem> items = DatabaseManager.UGCMarketItems.FindAll();
+        foreach (UGCMarketItem item in items)
+        {
+            AddListing(item);
+        }
+
+        Sales = new();
+        List<UGCMarketSale> sales = DatabaseManager.UGCMarketSales.FindAll();
+        foreach (UGCMarketSale sale in sales)
+        {
+            AddSale(sale);
+        }
+    }
+
+    public void AddListing(UGCMarketItem item)
+    {
+        Items.Add(item.Id, item);
+    }
+
+    public void RemoveListing(UGCMarketItem item)
+    {
+        Items.Remove(item.Id);
+    }
+
+    public List<UGCMarketItem> GetItemsByCharacterId(long characterId)
+    {
+        return new(Items.Values.Where(b => b.SellerCharacterId == characterId).ToList());
+    }
+
+    public List<UGCMarketItem> GetPromoItems()
+    {
+        List<UGCMarketItem> items = Items.Values.Where(x => x.PromotionExpirationTimestamp > TimeInfo.Now() && x.Status == UGCMarketListingStatus.Active).ToList();
+        Random random = RandomProvider.Get();
+        items = items.OrderBy(x => random.Next()).ToList();
+        return items.Take(12).ToList(); // 12 being the max the shop can display
+    }
```

Comment:
I feel this is easier to read. You can be the judge
```suggestion
    public List<UGCMarketItem> GetPromoItems()
    {
        return Items.Values.Where(x => x.PromotionExpirationTimestamp > TimeInfo.Now() && x.Status == UGCMarketListingStatus.Active)
            .OrderBy(_ => RandomProvider.Get().Next()).Take(12).ToList(); // 12 being the max the shop can display
    }
```